Gems/Atom/Asset/ImageProcessingAtom/Code/Source/Processing/ImageObjectImpl.cpp

```diff
@@ -352,6 +353,63 @@ namespace ImageProcessingAtom
         return (float)avg;
     }
 
+    AZ::Color CImageObject::ComputeLinearAlphaWeightedAverageColor() const
+    {
+        // Check if we are in a compressed or non-linear (sRGB) space
+        if (!CPixelFormats::GetInstance().IsPixelFormatUncompressed(m_pixelFormat)
+            || HasImageFlags(EIF_SRGBRead))
+        {
+            IImageObjectPtr clone(Clone(1)); // Only most detailed mip is needed
+            IImageObjectPtr linear = GetUncompressedLinearImage(clone);
+            return linear->ComputeLinearAlphaWeightedAverageColor();
+        }
+
+        // We are already in a linear, uncompressed space
+        AZ_Assert(CPixelFormats::GetInstance().IsPixelFormatUncompressed(m_pixelFormat),
+                "Expected an uncompressed image")
+        AZ_Assert(!HasImageFlags(EIF_SRGBRead), "Expected a linearly-encoded image")
+
+        // Accumulate weighted pixel colors and alpha of the top mip
+        double weightedSum[3] = { 0.0, 0.0, 0.0 };
```

Comment:
Any reason why you aren't using `AZ::Color` here and same for `color`? I think that would make it more obvious.